code/modules/reagents/chemistry/reagents/toxins.dm

```diff
@@ -1228,16 +1228,18 @@
 	color = "#20324D" //RGB: 32, 50, 77
 	metabolization_rate = 0.2
 	var/shock_timer = 0
+	var/normal_teslium = TRUE //Is this the base teslium, or the blob teslium. Used so it can be a subtype of teslium without inheriting the random shock from normal teslium as well as it's own
 	process_flags = ORGANIC | SYNTHETIC
 	taste_description = "electricity"
 
 /datum/reagent/teslium/on_mob_life(mob/living/M)
-	shock_timer++
-	if(shock_timer >= rand(5,30)) //Random shocks are wildly unpredictable
-		shock_timer = 0
-		M.electrocute_act(rand(5, 20), "Teslium in their body", 1, SHOCK_NOGLOVES) //Override because it's caused from INSIDE of you
-		playsound(M, "sparks", 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
-	return ..()
+	if(normal_teslium)
```

Comment:
Since blob teslium will not be able to pass this check, the `return ..()` is never reached and 
`/datum/reagent/proc/on_mob_life` will not be called for it. From what I can tell, this means the reagent never decays inside the mob, which is bad obviously. You'll probably want to move the `return ..()` left a tab, so it's always called.

Reply:
Yeah, that might be a good idea